markdownhighlighter.h

```diff
@@ -27,6 +31,25 @@ QT_END_NAMESPACE
 class MarkdownHighlighter : public QSyntaxHighlighter {
     Q_OBJECT
 
+#ifdef QT_QUICK_LIB
+    Q_PROPERTY(QQuickTextDocument *textDocument READ textDocument WRITE
+                   setTextDocument NOTIFY textDocumentChanged)
+
+    QQuickTextDocument *m_quickDocument = nullptr;
+
+   signals:
+    void textDocumentChanged();
+
+   public:
+    inline QQuickTextDocument *textDocument() const { return m_quickDocument; };
+    void setTextDocument(QQuickTextDocument *textDocument) {
+        if (!textDocument) return;
+        m_quickDocument = textDocument;
+        setDocument(m_quickDocument->textDocument());
+        Q_EMIT textDocumentChanged();
+    };
+#endif
+
```

Comment:
If QtQuick was found, add a `textDocument` property to allow usage in/with QML and using the `QTextDocument` from QML's `TextEdit`